annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastPackageResolver.java

```diff
@@ -53,4 +53,66 @@ private static String getParentPackage(String packageString) {
         }
         return packageString.substring(0, lastDotIndex);
     }
+
+    public void addFieldToFastType(Element element) {
+        FastFieldElement fastField = new FastFieldElement(element);
+        FastTypeElement fastType = getFastTypeForField(element);
+        if (fastType != null) {
+            fastType.addFastField(fastField);
+        } // otherwise ignore @FastFields if class is not annotated with @FastType
+    }
+
+    private FastTypeElement getFastTypeForField(Element element) {
+        Element ownerFastType = element.getEnclosingElement();
+        for (FastPackageElement fastPackage : cache.values()) {
+            for (FastTypeElement fastType : fastPackage.getFastTypes()) {
+                if (fastType.getElement().equals(ownerFastType)) {
+                    return fastType;
+                }
+            }
+        }
+        return null;
+    }
+
+    void inheritFields() {
+        HashSet<FastTypeElement> fastTypes = getAllFastTypes();
+        for (FastTypeElement fastType : fastTypes) {
+            Element currentType = fastType.getElement();
+            while (true) {
+                currentType = getSuperClass(currentType);
+                if (currentType == null) {
+                    break;
+                }
+                ArrayList<FastFieldElement> fastFieldElements = new ArrayList<>(getFastFieldsFor(currentType));
+                fastFieldElements.stream()
+                    .filter(fastField -> !isOverrideMethod(fastType.getFastFields(), fastField.getMethodName()))
+                    .forEach(fastType::addFastField);
+            }
+        }
+    }
+
+    private boolean isOverrideMethod(List<FastFieldElement> fastFields, String methodName) {
+        return fastFields.stream().anyMatch(fastField -> fastField.getMethodName().equals(methodName));
```

Comment:
It's not enough. Parameters should be the same. Moreover Java may still treat method as overridden if parameter is of brother type (need to check). You may skip this check now, but add ticket to Redmine to resolve it in the future.